packages/batch-tx-handler/src/l1-batch-handler/utils.ts

```diff
@@ -57,26 +74,13 @@ const getNextSerializedTransactionSize = (remainData: Uint8Array, start: number)
   return size;
 };
 
-export const getAllL2Msgs = (l2segments: Uint8Array[]): Uint8Array[] => {
-  const l2Msgs: Uint8Array[] = [];
-
-  for (let i = 0; i < l2segments.length; i++) {
-    const kind = l2segments[i][0];
-    let segment = l2segments[i].subarray(1);
-    /**
-     * Here might contain Timestamp updates and l1 block updates message here, but it is useless
-     * in finding tx hash here, so we just need to find tx related messages.
-     */
-    if (kind === BatchSegmentKindL2Message || kind === BatchSegmentKindL2MessageBrotli) {
-      if (kind === BatchSegmentKindL2MessageBrotli) {
-        segment = brotli.decompress(Buffer.from(segment));
-      }
-      l2Msgs.push(segment);
-    }
-    if (kind === BatchSegmentKindDelayedMessages) {
-      //TODO
-    }
-  }
+export const getAllL2Msgs = async (
+  l2segments: Uint8Array[],
+  afterDelayedMessagesRead: number,
+): Promise<Uint8Array[]> => {
+  let l2Msgs: Uint8Array[] = [];
+
+  [l2Msgs] = await extractL2Msg(l2segments, afterDelayedMessagesRead - 1, 0);
```

Comment:
Why don't we also take `afterDelayedMessagesRead` into account? (i.e., why do we do "-1" there?